Initial outline of NSSP vignette

kaitejohnson opened the pull request:
## Description

This PR closes #197. It makes an outline of the planned NSSP vignette, which will show pre-processing on simulated line list data and then will use a larger simulated reporting triangle to demonstrate the baselinenowcast workflow. 

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

<!-- Thanks again for this PR - @epinowcast dev team -->

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,44 @@
+#'   specified reference date and reported on the report date.}
+#' }
+#' @source Created for package demonstration.
+"mock_long_df"
```
also usually people call this synthetic data not mock so I would go with syn_nssp_what_kind_of_data_this_is

kaitejohnson commented:
> I started looking at this as it was marked for review but as I haven't been pinged shall I stop?

I was going to address code rabbit first... but I will address what you have so far, its still a lot bit rough

kaitejohnson commented on R/data.R:
```diff
@@ -0,0 +1,44 @@
+#' }
+#' @source Created for package demonstration to provide an example of how to
+#' pre-process this dataset to obtain a reporting triangle.
+"NSSP_line_list_raw"
```
No -- I just wanted to indicate in some way that this is what you would get straight from the API pull... will move to using mock as well though and make snake case

seabbs commented on DESCRIPTION:
```diff
@@ -12,6 +12,11 @@ Authors@R:
+    person(given = "Emily",
+           family = "Tyszka",
+           email = "etsyzka@umass.edu",
+            role = c("ctb"),
```
Did you mean to add her as an Author?

kaitejohnson replied:
Because they asked to hold off until they got permission. I can change to author.

seabbs commented on NEWS.md:
```diff
@@ -1,5 +1,6 @@
 # baselinenowcast 0.0.0.1000
 
+-   Add a vignette which demonstrates how to pre-process NSSP data and use `baselinenowcast` to nowcast cases of a specific syndromic surveillance.
```
What is NSSP data.

seabbs replied:
I would reword start with sss with NSSP in full as the case study

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,44 @@
+#'   specified reference date and reported on the report date.}
+#' }
+#' @source Created for package demonstration.
+"syn_nssp_rep_tri_df"
```
Can we just call this `syn_nssp_df`?

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,44 @@
+#'   YYYY-MM-DD HH:MM:SS format}
+#'   \item{c_race}{Character string indicating the race/ethnicity of the
+#'   patient}
+#'   \item{sex}{Character string indicating the gender of the patient}
```
Is this accurate? Aren't gender and sex different definitions?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+---
+title: "Nowcasting U.S. National Syndromic Surveillance Program (NSSP) data"
+description: "A nowcasting example applied to data from NSSP"
```
as in the news suggest refocus sss data using .,... (NSSP). 

Need to change title of the Rmd to use lower case nssp

kaitejohnson replied:
news suggest refocus sss ?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+
+# Introduction
+
+In this vignette, we will show an example of how to clean and pre-process line-list patient-level data from the [National Syndromic Surveillance](https://www.cdc.gov/nssp/index.html) (NSSP) dataset.
```
As above I would start with SSS and then NSSP as a specific example

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+
+In this vignette, we will show an example of how to clean and pre-process line-list patient-level data from the [National Syndromic Surveillance](https://www.cdc.gov/nssp/index.html) (NSSP) dataset.
+The NSSP Emergency Department (ED) visit dataset is a dataset widely used by public health departments in the United States, representing many but not all counties in the country.
+This dataset is entirely fake, it represents a small number of synthetic records that mirror the NSSP update fields.
```
```suggestion
This dataset is entirely synthetic, it represents a small number of synthetic records that mirror the NSSP update fields.
```

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+head(count_df)
+```
+
+We have now generated data in the format that we need to use the `baslinenowcast` package, which requires a long tidy dataframe with incident case counts indexed by reference date and report date.
````
link here and earlier to the concept of long and tidy data frames

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+
+# Pre-processing of larger synthetic dataset
+
+We'll start by loading in the synthetic reporting triangle dataframe, which is also provided as package data.
```
info box. How was this simulated

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+
+## Exploratory data analysis to identify an appropriate maximum delay
+
+We will start by computing the reporting delay and performing an exploratory data analysis of the delays in this dataset.
```
why

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+cdf_delay
+```
+Based on this figure, we can set the maximum delay to be 25 days as this is where 95% of the cases appear to have been reported.
+We will also set the nowcast date as 30 days before the maximum reference date in the dataset, May 5th, 2026.
````
why

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+```
+
+## Format for `baselinenowcast`
+We'll start by generating a reporting triangle from the `syn_nssp_rep_tri_df`, after we've removed all of the reports from after the nowcast date.
````
link to this concept

here the df already being called rep tri is pretty confusing with the way the text is written

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+
+## Format for `baselinenowcast`
+We'll start by generating a reporting triangle from the `syn_nssp_rep_tri_df`, after we've removed all of the reports from after the nowcast date.
+We will need to fill in all the possible reference dates and report dates with delays up until the maximum delay.
```
why

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+## Specify the `baselinenowcast` model
+
+We will use the helper function `allocate_reference_times()` to determine how many reference times to use for delay estimation and uncertainty estimation.
+We will use the defaults for this function, which uses 3 times the maximum delay as the total amount of training volume, split evenly between reference times used for delay estimation and reference times used for producing retrospective nowcasts for uncertainty estimation.
```
suggert linking to wher ethe defaults are defined vs writing out again as its more technical debt to try and keep in sync

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+```
+
+# Run the `baselinenowcast` workflow
+Here we demonstrate the workflow using the intermediate-level functions. For more details, see the [Getting Started](baselinenowcast.html) vignette.
````
Not sure this will mean much to a new user I suggest rephrasing i.e move to the end and say there are lower level functions and link

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
+```
+
+Next, we estimate and apply uncertainty by generating retrospective nowcasts and comparing against later observed data.
````
tell me about or link (ideally link) to some of the basic assumptions being made here (and above)

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,510 @@
+  rep_tri,
+  n_history_delay = n_history_delay,
+  n_retrospective_nowcasts = n_retrospective_nowcasts,
+  draws = 100
```
I'd suggest either setting this higher and filtering with a note. for ploitting or warning here we don't want users to actually use this few samples

kaitejohnson replied:
going to use default of 1000

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+```
+
+# Summarise and plot the nowcast
+Now that we have a dataframe of probabilistic nowcast draws for all reference times in our original reporting triangle, we can summarise the median and 50th and 95th quantiles.
````
might want to rewrite this with the audience in mind will this mean anything to them

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+  )
+```
+
+Next, we will join the initially reported data and the final evaluation data, both summarised by reference time.
````
Why?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+```
+
+# Plot against later observed "final" data
+Next we will make a plot of the nowcast
````
I dont think we hav etold anyone what a nowcast is or why they need one. Or visualised the data with truncation until now?

kaitejohnson replied:
have added at beginning and throughout this section

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+  ggtitle("Comparison of cases of ARI as of the nowcast date, later observed,\n and generated as a probabilistic nowcast") # nolint
+```
+</details>
+```{r}
````
Soo... is this good am I done? what do I do now? 

needs some kind of comment some kind of summary links to more resourcesx

seabbs reviewed (commented):
Core content seems to be mostly there. I suggest reviewing the writing with the audience in mind. Some specific comments.

kaitejohnson commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,510 @@
+```
+
+## Expand the data so that each "event" has its own column
+We will create two datasets which parse the characters in the columns `DischargeDiagnosisMDTUpdates` and `DischargeDiagnosisUpdates` , which contain a string listing the time stamp and diagnosis codes (respectively) of each "event" in the clinical encounter, formatted as:
````
There is probably a more efficient way to do this then creating two datasets and merging it *if* the data is perfectly aligned with the event numbers  -- but this is a bit "safer"? I think

seabbs commented:
Just whilst I am here the summary should likely go: 

Why we are doing this 

What we do in this vignette 

not the other way around. 

Also to "clean" is fairly meaningless better to say we go one blah blah data format to blah blah ddata format as this allows us to blah.,

kaitejohnson commented:
> I'm confused if you want me or rabbit to review. I updated my notification pings to wait for being asked for a review i.e top right?

Will swap back to draft and ping you specifically.

kaitejohnson commented:
> Just whilst I am here the summary should likely go:
> 
> Why we are doing this
> 
> What we do in this vignette
> 
> not the other way around.
> 
> Also to "clean" is fairly meaningless better to say we go one blah blah data format to blah blah ddata format as this allows us to blah.,

I think we need:
- Why we are doing this
- About the data (broadly --> this dataset) 
- What we do in this vignette ( a sentence at the end of about the data?) 

I don't think "What we do in this vignette" needs its own section as we want to save the details of what we are doing to describe alongside the code that does the thing

seabbs commented:
I meant the structure of the opening few sentences

kaitejohnson commented:
Updates made since @seabbs last review:
- rewrote the introduction to motivate why we want trends in real-time syndromic surveillance and why we need to nowcast it, then a description of the data (borrowed from original and specific revisions in language from Laura/Rosa)
- moved the details of the data transformations in the intro to be alongside when we do them in the code
- added descriptions of why we need the data in these different formats to ultimately compute delays for individual visits
- added a summary pointing to the baselinenowcast paper and other vignettes within the package
- addressed all comments (to best of my ability)

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,54 @@
+#' A synthetic dataset containing the number of incident cases indexed by
+#'   reference date and report date. While data of this form could be from any
+#'   source, this data is meant to represent the output of pre-processing the
+#'   `syn_nssp_line_list` dataset, which is a synthetic patient-level line list
```
use a markdown link to connect in pkgdown

kaitejohnson commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+---
```
I did change this hmm looks like it didn't update though.. will change the name fully then change it back and see if that works

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+## About syndromic surveillance system data
+
+Syndromic surveillance system data contains information at the visit-level on the timing and nature of the patient's clinical interactions.
+In this case study, we will use synthetic data from the United States' [National Syndromic Surveillance Program](https://www.cdc.gov/nssp/index.html) (NSSP) dataset available for participating jurisdictions through the CDC's ESSENCE platform, to nowcast cases of Broad Acute Respiratory Incidence (BAR).
```
CDC not defined

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+Clinical encounters may begin to be associated with diagnoses codes at different points in the patient or clinical processing journey (e.g. during registration, triage, clinical encounter, after laboratory results are returned, or during coding for billing, etc.), and are captured through update messages to the syndromic surveillance system once they are entered into a facility's electronic healt ...
+The difference between the visit date - when the patient registers in the emergency department- and the time of the diagnosis update pertaining to the diagnosis of interest, is used to compute a reporting delay for each patient.
+Reporting delays for diagnoses of interest can vary by a range of factors including by pathogen/syndrome, season, time of day or week, means of diagnosis, the electronic health record system, or treating facility.
+In this vignette, we will demonstrate how to use syndromic surveillance system data, particularly the timestamped data captured in each NSSP ESSENCE visit record, to produce nowcasts of cases of a particular diagnosis of interest.
```
maybe make it clearer people can changer the diagnosis codes

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+# Set seed for reproducibility
+set.seed(123)
+```
+
````
Land of the random spaces - space police - you have random spaces

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+Clinical encounters may begin to be associated with diagnoses codes at different points in the patient or clinical processing journey (e.g. during registration, triage, clinical encounter, after laboratory results are returned, or during coding for billing, etc.), and are captured through update messages to the syndromic surveillance system once they are entered into a facility's electronic healt ...
+The difference between the visit date - when the patient registers in the emergency department- and the time of the diagnosis update pertaining to the diagnosis of interest, is used to compute a reporting delay for each patient.
+Reporting delays for diagnoses of interest can vary by a range of factors including by pathogen/syndrome, season, time of day or week, means of diagnosis, the electronic health record system, or treating facility.
+In this vignette, we will demonstrate how to use syndromic surveillance system data, particularly the timestamped data captured in each NSSP ESSENCE visit record, to produce nowcasts of cases of a particular diagnosis of interest.
```
Here or earlier tell me qabout the structure of this vignette. i.e. basically the subtitles

seabbs replied:
I think this should go earlier and this statement is redundant as the section is meant to be background info

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+<div class="alert alert-warning" role="alert">
+<strong>Note:</strong> This dataset does not represent data from real patients, it is entirely synthetic and designed to mirror the NSSP update fields, which records the timing of a clinical encounter and the corresponding diagnoses code, if any.
+</div>
+
```
space

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+
+## Define the "syndrome" definition
+
+In this near-real-time emergency department dataset, public health surveillance, or "syndrome" definitions have been developed for a range of public health concerns. These definitions typically rely on the presence of diagnosis code(s), specific free text captured in clinical notes, or a combination of these.
```
have been developed? Quite abstract phrasing by who? the field?

kaitejohnson replied:
I actually don't know .... good question for Laura/Rosa unless you know this?

seabbs replied:
No I don't know either. I would guess the field or by those heroic "practioners" we like to talk about

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+In some instances, exclusion criteria are incorporated into these definitions to improve their specificity.
+
+In this example, we will focus on syndromic surveillance definitions that rely solely on the presence of one or more diagnoses codes, as this means that patients tend to transition from a non-case to case only once (rather than reverting back to a non-case during the course of their clinical encounter).
+Our nowcasting methods primarily focus on the former use-case, though extensions to be able to account for downward revisions as cases transition from case to non-case, are planned for future work.
```
I think you should move this to later when we actually nowcast. It also is more nuanced than this as it can already be done without an extension. I would say when the point in the analysis happens that makes this assumption

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+  syn_nssp_diagnoses_wide,
+  matches("DischargeDiagnosisUpdates")
+)), 1)
+syn_nssp_diagnoses_long <- syn_nssp_diagnoses_wide |>
```
if this is literally identical code I would suggest it is more readable to define a function say why and then use it twice

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+
+Next, we will clean up the time stamps in the data so that the `time_stamp` column is formatted as `%Y-%m-%d %H:%M:%S`, and then we will filter out an events that are not present (updates are NAs).
+```{r}
+syn_nssp_time_stamps_long <-
````
overwriting the same names here I would suggest not doing so

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+  drop_na() |>
+  select(event_id, diagnoses_codes)
+```
+
````
space police

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+  syn_nssp_diagnoses_long,
+  by = "event_id"
+) |>
+  filter(diagnoses_codes != ";;|") # drops empty updates
```
in general in line comments despite being the love of LLMs lives are quite distracting

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+```{r}
+nssp_updates <- nssp_merged |>
+  mutate(arrival_to_update_delay = as.numeric(difftime(
+    as.POSIXct(time_stamp), as.POSIXct(C_Visit_Date_Time),
````
I would suggest on pass doing a variable transform to posixct and then using that rather than having to keep writing this as it makes the code long and boring to read

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+head(clean_line_list)
+```
+
+For nowcasting, we want to compute the number of incident cases indexed by reference and report date, so we can aggregate by reference and report date and compute the delay distribution.
````
maybe this needs a subsection to make it clear ?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+  group_by(reference_date, report_date) |>
+  summarise(count = n()) |>
+  mutate(delay = as.integer(report_date - reference_date))
+```
````
space needed. Suggest asking coderabbit to help police this in the future

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+```
+Looking at this data, we can see that there is one case where there is a negative delay, which indicates that the time stamp of the diagnosis update was recorded before the start of the visit.
+This is likely due to a data entry error.
+We could choose to code these as having 0 delay, but if they are indeed erroneous this might bias our delay distribution estimation towards zero when they are equally likely to have had longer delays.
````
reword. It depends on the data generating mechanisms for these. In some settings they ar true negatives and so the closest imputation is to set as zero. If you remove as suggested here in those cases then it actually leads to more bias (longer delays). 

I think this is fine here given the reason but I would nuance around above point and data experts knowing their data blah

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+head(count_df)
+```
+
+We have now generated data in the format that we need to use the `baselinenowcast` package, which requires a [long tidy dataframe](https://r4ds.had.co.nz/tidy-data.html) with incident case counts indexed by reference date and report date.
````
link to something in epinowcast about the dataformat we nmeed and why maybe getting started etc.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+## Specify the `baselinenowcast` model
+
+We will use the helper function `allocate_reference_times()` to determine how many reference times to use for delay estimation and uncertainty estimation, using the package defaults.
+See the "Default Settings" section of the [mathematical model](model_definition.html) vignette for more details.
```
Comment on what this means or something print what we get

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+First, we generate the point estimate of the nowcast as a "filled in" reporting triangle.
+See the sections "Estimating the delay distribution from a reporting triangle" and "Point nowcast generation" of the [mathematical model](model_definition.html) for more details.
+```{r}
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
````
print

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+
+# Summarise and plot the nowcast
+
+Now that we have a dataframe of probabilistic nowcast draws, which represent our predicted "final" case count at each reference time, we can summarise them to compute the 50th and 95th percent prediction intervals, which are used to communicate uncertainty in the nowcast.
```
Better structure:" Human level what we have - these are knwon as ,.... not jargon followed by def.

seabbs replied:
also can be used to summarise uncertainty not communicate not these are arbitary and we would suggest showing more prediction intervals  if possible.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+  )
+```
+
+Next, we will join the initially reported cases and the held out, true final cases to our probabilistic nowcast.
````
Why? Same points. 

We need to do X therefore we will do why. No one cares about technical detail for its own sake we need to justify every step first.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+Next, we will join the initially reported cases and the held out, true final cases to our probabilistic nowcast.
+The final cases will be used to evaluate the performance of our nowcast given the right-truncated initial reports.
+```{r}
+training_df_by_ref_date <- training_df |>
````
too much in one chunk. Too much repeated code def function and use.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+head(nowcast_w_data)
+```
+
+## Plot against later observed "final" data
````
space also several above. Use LLM maybe get a precommit hook

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+## Plot against later observed "final" data
+Next we will make a plot of the nowcasted "final" cases compared to the initially reported, right-truncated cases and the true "final" reports.
+```{r}
+combined_data <- nowcast_w_data |>
````
Why am I being shown this code if no one is telling me what it does?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+```
+
+## Plot against later observed "final" data
+Next we will make a plot of the nowcasted "final" cases compared to the initially reported, right-truncated cases and the true "final" reports.
````
Tell me about what I see is this good. Is this useful for public health why. What would happen if I didn't do it. Link back to the need we talked about at the beginning. 

I think we need to visualise the data earlier to highlight the problem as people may have died from boredom at this point. Suggest visualising after cleaning as a motivation for doing nowcasting as at the moment its tell not show which is bad

kaitejohnson replied:
Ok I added a plot of the initial data in red right after we create `training_df` which shows the rapidly declining recent trends

kaitejohnson replied:
Plus motivation etc for nowcasting

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+library(tidyr)
+library(readr)
+library(stringr)
+library(lubridate)
```
are you actually using lubridate now?

kaitejohnson replied:
Yes for `days(delay)`

seabbs replied:
I think you may be able to avoid that if you would lke it

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+From there we used the `baselinenowcast` workflow to estimate a delay distribution, apply it to generate a point nowcast, and estimate and apply uncertainty to generate probabilistic nowcasts.
+As a final step, we compared our nowcasts of the eventual final observed case counts to what we later observed and the right-truncated initial reports.
+
+Next steps include scoring the nowcasts we generated using proper scoring rules such as the weighted interval score or the continuous ranked probability score and found the coverage metrics.
```
why are you teasing me give me some links it doesn't help me knowing you know how to do this I want to be able to do this.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+As a final step, we compared our nowcasts of the eventual final observed case counts to what we later observed and the right-truncated initial reports.
+
+Next steps include scoring the nowcasts we generated using proper scoring rules such as the weighted interval score or the continuous ranked probability score and found the coverage metrics.
+Since the `baselinenowcast` method is very fast, we encourage users to test out on a range of nowcast dates, and consider other model specifications such as weekday specific delay estimation or variations in the amount of training data used for delay and uncertainty estimation.
```
```suggestion
We encourage users to test out on a range of nowcast dates, and consider other model specifications such as weekday specific delay estimation or variations in the amount of training data used for delay and uncertainty estimation.
```

seabbs replied:
Why do you encourage users to do this? You have nothing to base this on./ 

I suggest you do more on the nowcast results i.e look at ways it might go wriong etc then link to this when suggesting this. Also need to signpost what in the code they would need to change and how. How for example do they filter to look at different time points exactly?

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+Since the `baselinenowcast` method is very fast, we encourage users to test out on a range of nowcast dates, and consider other model specifications such as weekday specific delay estimation or variations in the amount of training data used for delay and uncertainty estimation.
+For more details on the performance of these method specifications applied to data from COVID-19 in Germany and norovirus cases in England, see our latest [pre-print](https://www.medrxiv.org/content/10.1101/2025.08.14.25333653v1.full).
+
+We encourage users to explore model specifications tailored to their data and scenario, which can be done most readily via the modular low level function interface described in detail in the [Getting Started](baselinenowcast.html) vignette.
```
this before the paper. Link to the model def here.

seabbs commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+
+Next steps include scoring the nowcasts we generated using proper scoring rules such as the weighted interval score or the continuous ranked probability score and found the coverage metrics.
+Since the `baselinenowcast` method is very fast, we encourage users to test out on a range of nowcast dates, and consider other model specifications such as weekday specific delay estimation or variations in the amount of training data used for delay and uncertainty estimation.
+For more details on the performance of these method specifications applied to data from COVID-19 in Germany and norovirus cases in England, see our latest [pre-print](https://www.medrxiv.org/content/10.1101/2025.08.14.25333653v1.full).
```
rewrite. Its not clear that performance is context specific. Point should be. We think our defaults are fairly robust but its important to tune for your data set and needs. 
using our tools you can easily do blah and you can also do blah blah blah using modular components and your own code. 

In blah we show excamples of us tuning the method for blah contexts and evaluate the results of these tunings we also have open code link that you might want to use if interested in doing something simuilar for your own problem

seabbs reviewed (changes requested):
Nearly there. Much improved. I think we should work out how to automate the space checking. Key point is aim for show and not tell communication at all times. I have pointed out a few specific instances of tell not show.

kaitejohnson commented on vignettes/NSSP_nowcast.Rmd:
```diff
@@ -0,0 +1,539 @@
+    reference_date = reference_date,
+    report_date = seq(reference_date, min(
+      reference_date + max_delay,
+      nowcast_date
```
Have edited this code and described each step

kaitejohnson commented on vignettes/NSSP_nowcast.Rmd:
````diff
@@ -0,0 +1,539 @@
+head(rep_tri_df)
+```
+
+As a final pre-processing step, we'll format the reporting triangle as just a matrix, for compatibility with the low-level functions in `baselinenowcast`.
````
edited the above to explain we want 0s for observations that occurred and we didn't have data vs NAs for those that have yet to be observed
